tests/kafkatest/tests/replication_test.py

```diff
@@ -15,45 +15,86 @@
 
 from ducktape.tests.test import Test
 from ducktape.utils.util import wait_until
+from ducktape.mark import matrix
+from ducktape.mark import parametrize
 
 from kafkatest.services.zookeeper import ZookeeperService
 from kafkatest.services.kafka import KafkaService
 from kafkatest.services.verifiable_producer import VerifiableProducer
 from kafkatest.services.console_consumer import ConsoleConsumer
 
-import signal
 import time
 
 
+"""
```

Comment:
This is sort of a docstring but a) not associated with anything and b) doesn't give any context.

I think you either want to move this up to be the module's docstring (top of the file) or move this into the class's docstring. The class's docstring also provides some context.